src/traces/bar/plot.js

```diff
@@ -511,17 +498,31 @@ function appendBarText(gd, plotinfo, bar, cd, i, x0, x1, y0, y1, r, overhead, op
             textHeight = textBB.height;
 
             var textHasSize = (textWidth > 0 && textHeight > 0);
-            var fitsInside = (textWidth <= barWidth && textHeight <= barHeight);
-            var fitsInsideIfRotated = (textWidth <= barHeight && textHeight <= barWidth);
-            var fitsInsideIfShrunk = (isHorizontal) ?
-                (barWidth >= textWidth * (barHeight / textHeight)) :
-                (barHeight >= textHeight * (barWidth / textWidth));
-
-            if(textHasSize && (
-                fitsInside ||
-                fitsInsideIfRotated ||
-                fitsInsideIfShrunk)
-            ) {
+
+            var fitsInside;
+            if(barIsRounded) {
```

Comment:
@archmoj @alexcjohnson What I've done here is:

- Factor the existing `fitsInside` logic out into a separate function
- If the bar is not rounded, just call that function
- If the bar is rounded:
  - Imagine the bar is shaped like a rectangle with squares of side length `r` chopped out at each corner (so a chunky cross shape)
  - Then, the text can fit either in the "wide" direction or the "tall" direction (this is similar to the "wide text" and "tall text" cases for the text scaling)
  - Use the `textFitsInsideBar` function to check each of those cases. The text fits if either case is true.

It's a pretty rough approximation but seemed better than calling the `scaleTextForRoundedBar` function multiple times.